### ISSUE TITLE
[Feature request] Conditional loading of SSH keys

### ISSUE THREAD

**Issue by schklom** [dlech/KeeAgent#390]
I manage SSH keys per client. So my laptop uses one SSH key for all purposes, same for my stationary computers.
When I am on my laptop, I want my laptop's SSH key to load. Same when I am on other computers.

I have not found a way to do that on KeeAgent. One machine uses Windows, the other uses Linux, so I can't put the same path for both.

KeePass implemented this condition to allow opening child databases depending on the hostname: https://sourceforge.net/p/keepass/discussion/329220/thread/509d35a111

**Comment by dlech** [comment]
> KeePass implemented this condition to allow opening child databases depending on the hostname

It sounds like this feature could solve your issue.

Another solution could be to use environment variables in the path of an external key file.

**Comment by schklom** [comment]
> It sounds like this feature could solve your issue.

Exactly, but this is only for the plugin KeeAutoExec, hence this feature request :)

> Another solution could be to use environment variables in the path of an external key file.

I guess this will do in the meantime.

**Comment by dlech** [comment]
> Exactly, but this is only for the plugin KeeAutoExec, hence this feature request :)

It seems like you can use both extensions together to do what you want. If you use the KeeAutoExec extension to load a per-host database, you can store the SSH keys in that database to get per-host keys.

**Comment by schklom** [comment]
Not ideal, but it should work. Thanks for the good idea!
I think I should keep the issue open because it was not really solved, but feel free to close it if you prefer :)

**Comment by spali** [comment]
Would have an additional use-case for this feature.
My problem is most of the time, I have a too many auth error due to many key get tried.
I use keeagent in multiple occasion with different devices. Most, if not all keys can are occasion specific.
But to use it with multiple devices I can't use the approach with KeeAutoExec and environment variables like hostname, as I don't have a environment variable that is reliable enough to have a unique value per occasion but on all devices.
I thought on the openssh constraint feature, but I have keys that are used to connect to hundred of virtual machines. So cant list them all... the only thing that would work would be username only... but thats not supported as openssh wants a host and I can't list all hosts as the names and keys are always changing.

Currently I have to load the keys manually on occasion when I start to work. A predefined set to load would help a lot or better a predefined set (or keys) based on hostname. This way I only need to maintain a list of hosts per occasion (keyset).
